mlflow/models/utils.py

```diff
@@ -466,14 +466,17 @@ def _reshape_and_cast_pandas_column_values(name, pd_series, tensor_spec):
         )
 
     if np.isscalar(pd_series[0]):
-        if tensor_spec.shape != (-1,):
-            raise MlflowException(
-                f"The input pandas dataframe column '{name}' contains scalar "
-                "values, which requires the shape to be (-1,), but got tensor spec "
-                f"shape of {tensor_spec.shape}.",
-                error_code=INVALID_PARAMETER_VALUE,
-            )
-        return _enforce_tensor_spec(np.array(pd_series, dtype=tensor_spec.type), tensor_spec)
+        for shape in [(-1,), (-1, 1)]:
+            if tensor_spec.shape == shape:
+                return _enforce_tensor_spec(
+                    np.array(pd_series, dtype=tensor_spec.type).reshape(shape), tensor_spec
+                )
+        raise MlflowException(
+            f"The input pandas dataframe column '{name}' contains scalar "
```

Comment:
We should also update the error message here if we make this change.

Reply:
```suggestion
            f"The input pandas dataframe column '{name}' contains scalar or one dimension array "
```

Reply:
@WeichenXu123 We check `np.isscalar(pd_series[0])` on line 468, the column doesn't contain one dimension array.

Reply:
oh, yes. you are right.